script/split_tests.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python3
+"""Helper script to split test into n buckets."""
+
+from __future__ import annotations
+
+import argparse
+from dataclasses import dataclass, field
+from math import ceil
+import os
+import subprocess
+import sys
+
+
+class Bucket:
+    """Class to hold bucket."""
+
+    def __init__(
+        self,
+    ):
+        """Initialize bucket."""
+        self.tests = 0
+        self._paths = []
+
+    def add(self, part: TestFolder | TestFile):
+        """Add tests to bucket."""
+        self.tests += part.total_tests
+        self._paths.append(part.path)
+
+    def get_paths_line(self) -> str:
+        """Return paths."""
+        return " ".join(self._paths) + "\n"
+
+
+class BucketHolder:
+    """Class to hold buckets."""
+
+    def __init__(self, tests_per_bucket: int, bucket_count: int) -> None:
+        """Initialize bucket holder."""
+        self._tests_per_bucket = tests_per_bucket
+        self._bucket_count = bucket_count
+        self._current_bucket = Bucket()
+        self._buckets: list[Bucket] = [self._current_bucket]
+        self._last_bucket = False
+
+    def split_tests(self, tests: TestFolder | TestFile) -> None:
+        """Split tests into buckets."""
+        if (
+            self._current_bucket.tests + tests.total_tests < self._tests_per_bucket
+        ) or self._last_bucket:
+            self._current_bucket.add(tests)
+            return
+
+        if isinstance(tests, TestFolder):
+            for test in tests.children.values():
+                self.split_tests(test)
+            return
+
+        # Create new bucket
+        if len(self._buckets) == self._bucket_count:
+            # Last bucket, add all tests to it
+            self._last_bucket = True
+        else:
+            self._current_bucket = Bucket()
+            self._buckets.append(self._current_bucket)
+
+        # Add test to new bucket
+        self.split_tests(tests)
+
+    def create_ouput_file(self) -> None:
+        """Create output file."""
+        with open("pytest_buckets.txt", "w") as file:
+            for bucket in self._buckets:
+                print(f"Bucket has {bucket.tests} tests")
+                file.write(bucket.get_paths_line())
+
+
+@dataclass
+class TestFile:
+    """Class to hold number of tests."""
```

Comment:
The doc string can be better.

Something like this:

```suggestion
    """This class represents a single test file and its tests."""
```